### PR TITLE
Fix some windows archive issues

### PR DESCRIPTION
**Pull request by cbednarski**
- This fixes https://github.com/mitchellh/packer/issues/2165
- Includes some windows compatibility fixes and some new tests

### CODE REVIEW

File: archive/archive.go [hunk 1]
```diff
@@ -359,6 +376,8 @@ func copyDirWalkFn(
 func copyConcreteEntry(
 	tarW *tar.Writer, entry string,
 	path string, info os.FileInfo) error {
+	// Windows
+	path = filepath.ToSlash(path)
```
Comment (mitchellh) [inline archive/archive.go]: Can you add a newline after this? Nitpick

File: archive/vcs_test.go [hunk 2]
```diff
@@ -6,21 +6,40 @@ import (
...
 	oldName := filepath.Join(testDir, "DOTgit")
 	newName := filepath.Join(testDir, ".git")
+
+	cleanup := func() {
+		os.Rename(newName, oldName)
+		// Windows leaves an empty folder lying around afterward
+		if runtime.GOOS == "windows" {
+			os.Remove(newName)
+		}
+	}
```
Comment (mitchellh) [inline archive/vcs_test.go]: newline after here as well

**Comment by mitchellh** [comment]
Just a little style thing but looks great!

**Comment by mitchellh** [comment]
LGTM

File: archive/vcs_test.go (hunk 2 above)
Comment (cbednarski) [inline archive/vcs_test.go]: Done!

**Pull request merged**